### ISSUE TITLE
Researching Hi-Sec Auto Tracking does not increase its upgrade level

### ISSUE THREAD

**Issue by Joshua-Leibold** [Ahli/sc2xml#140]
Ingame: https://i.gyazo.com/044980e92e356199dfcf0f02f63b786a.gif

```xml
<CUpgrade id="HiSecAutoTracking">
     <EffectArray Reference="Weapon,AutoTurret,Level" Value="1"/>
     <EffectArray Reference="Weapon,LongboltMissile,Level" Value="1"/>
     <EffectArray Reference="Weapon,TwinIbiksCannon,Level" Value="1"/>
</CUpgrade>
```

After fix: https://i.gyazo.com/15c248a98b17543bf4f7120517efc782.gif

**Comment by Ahli** [comment]
I don't like non-damage upgrades altering the level label due to inconsistency.

For example, Hydralisk's Grooved Spines upgrades are not raising the level label and shouldn't.

**Comment by Joshua-Leibold** [comment]
You are correct that Hydralisk Grooved Spines should not raise the level upgrade. However, Grooved Spines's icon does not appear on the unit info panel so does not require a level upgrade effect


This is the Grooved Spines (research) icon: 
![b3b905cc3976f7e07912c99f8d05eb5d](https://github.com/Ahli/sc2xml/assets/143013216/e5bc8d0a-402a-487d-906c-3f4ec65c0782)

This is the Hydralisk (generic zerg ground missile) unit info panel icon: 
![d5ef01b1ccb7aece365885c501d2b04a](https://github.com/Ahli/sc2xml/assets/143013216/37b0dec3-0b24-45f5-8976-ab4306a1ffc6)

This is the Hi-Sec Auto Tracking (research) icon:
![961db5bedf58841b5b003dec1548b5eb](https://github.com/Ahli/sc2xml/assets/143013216/b2309be7-7fb3-482a-8db3-9c5c2c6b40eb)

This is in contrast to the missile turret which does use the dedicated Hi-Sec Auto Tracking icon on the unit info panel (the button is even called Hi-Sec auto tracking in the files: there is no "generic ranged building attack" icon) :
![34769fbc1c262ad2370f140cad2174ec](https://github.com/Ahli/sc2xml/assets/143013216/653e9b12-83fb-4eb3-b548-d326ddef7995)

If the spore crawler had a range upgrade, I would agree the level should not be increased since the spore crawler uses the ranged ground attack icon like the hydralisk and presumably that ranged upgrade would use a separate hypothetical icon. However, unless you're planning on changing the Terran structures to not share an icon with Hi-Sec Auto Tracking, this change keeps things intuitive and consistent.

Furthermore, if the upgrade level was not increased then Hi-Sec Auto tracking would have to be added to the list of upgrades for which units do not hold a passive icon for, creating another potential inconsistency which if rectified would cause this icon to have to appear in two separate places on the UI for the same unit.

I've also still got the more practical gameplay application on my side. There is currently no way to increase the upgrade level of the Hi-Sec Auto Tracking unit info panel icon by other means, making increasing the level unambiguous (one does not need to hover over the Hi-Sec icon to know what has changed, it will always signal the range as the upgrade research button describes). This in contrast to something like Grooved Spines or Resonating Glaives, which if the upgrade level increased the Zerg ground missile or Protoss Ground weapon upgrades would cause the level increase to be ambiguous. 

Very anecdotally: I have an embarrassing amount of times wondered myself if my mutas, medivacs etc. were getting hit by Hi-Sec auto tracking and I have mistakenly clicked on the turret to look at the unit info panel icon to check for upgrades like I would any other unit, seen no upgrade level on the attack, and incorrectly concluded the weapon was not modified

**Comment by Joshua-Leibold** [comment]
Resolved in 5.0.15
